### ISSUE TITLE
salt/grains/core.py calling powershell 

### ISSUE THREAD

**Issue by damon-atkins** [saltstack/salt#45181]
### Description of Issue/Question
```
C:\Salt-Dev>python.exe -B c:\Salt-Dev\salt\scripts\salt-call -l info test.ping
[INFO    ] cmdmod._run Executing command 'Powershell -NonInteractive -NoProfile "Get-WmiObject -class MSFC_FibrePortHBAAttributes -namespace \"root\WMI\" | Select -Expandproperty Attributes | %{($_.PortWWN | % {\"{0:x2}\" -f $_}) -join \"\"} | ConvertTo-JSON"' in directory 'C:\Users\root'
[ERROR   ] cmdmod.run Command 'Get-WmiObject -class MSFC_FibrePortHBAAttributes -namespace "root\WMI" | Select -Expandproperty Attributes | %{($_.PortWWN | % {"{0:x2}" -f $_}) -join ""} | ConvertTo-JSON' failed with return code: 1
[ERROR   ] cmdmod.run output: Get-WmiObject : Not supported
At line:1 char:1
+ Get-WmiObject -class MSFC_FibrePortHBAAttributes -namespace "root\WMI ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Get-WmiObject], ManagementException
    + FullyQualifiedErrorId : GetWMIManagementException,Microsoft.PowerShell.Commands.GetWmiObjectCommand
[ERROR   ] cmdmod.powershell Error converting PowerShell JSON return
Traceback (most recent call last):
  File "c:\Salt-Dev\salt\salt\modules\cmdmod.py", line 3268, in powershell
    return json.loads(response)
  File "c:\Python27\Lib\json\__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "c:\Python27\Lib\json\decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "c:\Python27\Lib\json\decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
local:
    True
```
### Setup
oxygen branch as of 2017-12-28

Their has been a fair amount of effort not to call powershell as part of salt starting, as its slows everything down.  This should be calling WMI direct from python. And to a lesser extent wmic 

```
./salt/grains/core.py:    ps_cmd = r'Get-WmiObject -class MSFC_FibrePortHBAAttributes -namespace "root\WMI" | Select -Expandproperty Attributes | %{($_.PortWWN | % {"{0:x2}" -f $_}) -join ""}'
./salt/grains/core.py:def _windows_iqn
./salt/modules/rdp.py:    rdp = ('$RDP = Get-WmiObject -Class Win32_TerminalServiceSetting '
```
core.py has already a lot of python based wmi call examples

Introduced in https://github.com/saltstack/salt/pull/43647 by @sdodsley not approved by  @saltstack/team-windows

Please label this with oxygen

**Comment by garethgreenaway** [comment]
Deferring to @cachedout and @Ch3LL for the Oxygen label.

**Comment by rallytime** [comment]
@damon-atkins This should be fixed with #45113.

**Comment by rallytime** [comment]
And actually this is a duplicate of #44894.

**Comment by damon-atkins** [comment]
Reopen please. Powershell call should be replaced with wmi python call.

**Comment by damon-atkins** [comment]
And if it must run power shell it should check first for Get-WmiObject existence before starting powershell if possible.

**Comment by rallytime** [comment]
@twangboy Can you comment here?

**Comment by damon-atkins** [comment]
If one powershell call is in here which takes 8 seconds, then more people will notice the easier way to add grains and more people will add them this way and the result will be salt-call taking a minute collect grains.

`salt.modules.cmdmod.shell_info`  was introduce primary to reduce the about of time taken to discover the version of powershell and its modules as it works it out without running `powershell` which myself and @twangboy and others worked on.  It would be a pity for salt-call and minion startup start being slow again.

**Comment by twangboy** [comment]
@damon-atkins @rallytime Trying to find a box with Fibre Channel on it

**Comment by twangboy** [comment]
Not stale
